Sources/ImageFetcher/ImageFetcher.swift

```diff
@@ -171,30 +184,25 @@ public extension ImageFetcher {
     /// Cancels an in-flight image load
     /// - Parameter imageConfiguration: The configuation of the image to be downloaded.
     func cancel(_ imageConfiguration: ImageConfiguration) {
-        guard let task = self[imageConfiguration] else {
+        guard let task = tasks[imageConfiguration.key], task.isPending else {
             return
         }
-
         task.cancel()
-        task.operation = nil
-        task.handler = nil
-
-        tasks.remove(task)
+        tasks[imageConfiguration.key] = nil
     }
 
-    subscript (_ url: URL) -> ImageFetcherTask? {
+    subscript(_ url: URL) -> ImageFetcherTask? {
         self[ImageConfiguration(url: url)]
     }
 
-    subscript (_ imageConfiguration: ImageConfiguration) -> ImageFetcherTask? {
-        tasks.first(where: { (task) -> Bool in
-            task.configuration == imageConfiguration
-        })
+    subscript(_ imageConfiguration: ImageConfiguration) -> ImageFetcherTask? {
+        tasks[imageConfiguration.key]
     }
 
     /// Deletes all images in the cache
-    func deleteCache() throws {
-        try cache.syncDeleteAll()
+    func deleteCache() async throws {
+        try await cache.deleteAll()
+        tasks = [:]
```

Comment:
Motivation: I was testing this in an app that used a fetcher to populate a collection view with images from JSONPlaceholder and I had a button wired up to `deleteCache()` with the expectation that that would be sufficient to reset the fetcher's state and fetch the images from the server again.

Would you suggest something along the lines of a separate `deleteMemoryCache()` method?

Reply:
Does a user need to reinitialize internal memory storage? What happens if a download is in flight?

Reply:
I can see the unconstrained growth of the memory cache being a problem in some situations so I think it would good to provide some way to manage that.

Re: in-flight requests - this is a good question and something that I don't handle sufficiently atm.

Reply:
I believe these two points are tied to each other. I also believe the way to manage that is already provided with `cancel(_:)` as it should cancel the task.

Another way for this to be managed is to add back max concurrent download operations which will allow the user to have better control based on their constraints.